tensorflow/compiler/mlir/xla/transforms/legalize_tf_patterns.td

```diff
@@ -491,6 +491,48 @@ def : Pat<(TF_EluGradOp AnyStaticShapeTensor:$gradients, AnyRankedTensor:$featur
                (HLO_ConstOp:$one (GetScalarOfType<1> $features)),
                (BinBroadcastDimensions $one, $features))))>;
 
+//===----------------------------------------------------------------------===//
+// Selu op patterns.
+//===----------------------------------------------------------------------===//
+
+def : Pat<(TF_SeluOp AnyRankedTensor:$features),
+          (HLO_SelectOp
+           (HLOClient_BroadcastCompareOp
+              $features,
+              (HLO_ConstOp:$zero (GetScalarOfType<0> $features)),
+              (BinBroadcastDimensions $zero, $features),
+              HLO_COMPARISON_DIRECTION_GT, (HLO_DEFAULT_COMPARISON_TYPE)),
+           (HLOClient_BroadcastMulOp
+            $features,
+            (HLO_ConstOp: $scale
+              (NativeCodeCall<"$_builder.getFloatAttr(((*features.begin()).getType()).cast<mlir::RankedTensorType>().getElementType(), 1.0507009873554804934193349852946)">)),
```

Comment:
we can define this outside to keep the pattern concise? See Get2DTransposePerm in this file. 

Also, use getElementTypeOrSelf helper.

Reply:
Done.